pkg/engine/renderer/dependencies.go

```diff
@@ -161,13 +171,35 @@ func sanitizeAndSerialize(origObj *unstructured.Unstructured) (string, error) {
 	// OwnerReferences need to be skipped as they contain a changing UID
 	obj.SetOwnerReferences([]metav1.OwnerReference{})
 
+	// These fields should not be present when we get the obj from our task or from the LastAppliedConfigAnnotation, but
+	// we may encounter objects that are not deployed by KUDO and therefore do not have the LastAppliedConfigAnnotation.
+	// This is a best effort attempt to make the hash as stable as possible
+	v := int64(0)
+	obj.SetResourceVersion("")
+	obj.SetClusterName("")
+	obj.SetFinalizers([]string{})
+	obj.SetContinue("")
+	obj.SetCreationTimestamp(metav1.Time{})
+	obj.SetDeletionGracePeriodSeconds(&v)
+	obj.SetGenerateName("")
+	obj.SetGeneration(0)
+	obj.SetManagedFields([]metav1.ManagedFieldsEntry{})
+	obj.SetRemainingItemCount(&v)
+	obj.SetSelfLink("")
+	obj.SetUID("")
+
+	// This may be set or not, depending if the original object is typed or unstructured. To be safe we nil it
+	obj.SetGroupVersionKind(schema.GroupVersionKind{})
+
 	// Annotations are notorious for containing quickly changing strings: plan/phase/task names, uids, dates, etc.
 	obj.SetAnnotations(map[string]string{})
 
 	return ToYaml(obj)
 }
 
-// resourceDependency returns the resource of type t with the given namespace/name, either from the passed in list of objects or the last applied configuration from the API server
+// resourceDependency returns the resource of type t with the given namespace/name, either from the passed in list of
```

Comment:
this comment doesn't make sense to me... "resource of type t"... there is no "t"

Reply:
Yeah, this was left over from an early implementation and wasn't adjusted when I refactored the code to use the `resourceDependency` struct. Fixed it.